tempodb/encoding/common/interfaces.go

```diff
@@ -51,3 +51,17 @@ func DefaultCompactionOptions() CompactionOptions {
 		OutputBlocks:       1,
 	}
 }
+
+type TraceIterator interface {
+	// Next returns the next trace and optionally the start and stop times
+	// for the trace that may have been adjusted.
+	Next(ctx context.Context) (ID, *tempopb.Trace, uint32, uint32, error)
```

Comment:
> move the []byte -> *Trace unmarshalling within the vparquet

To decode the []byte it would also need an `ObjectDecoder`, and move `Iterator` to `encoding/common`.  Agree this should be fine.

> what percentage of WAL traces actually go through the model/combiner

It looks like we don't have good metrics for this, hard to say and likely workload dependent (if spans arrive irregularly or after trace flush).